data/persistence/src/commonMain/kotlin/social/androiddev/common/persistence/localstorage/DodoAuthStorageImpl.kt

```diff
@@ -31,29 +34,31 @@ internal class DodoAuthStorageImpl(
      * The user can set up multiple accounts on their device. So we
      * key the AccessToken by the unique server/domain
      */
-    private var diskCache: Map<String, AccessToken>
+    private var diskCache: LinkedHashMap<String, AccessToken>
         get() {
             return settings
                 .getStringOrNull(KEY_ACCESS_TOKENS_CACHE)
                 ?.let { str ->
                     json.decodeFromString(ListSerializer(AccessToken.serializer()), str)
-                        .associateBy { it.server }
+                        .associateByTo(linkedMapOf()) { it.server }
                 }
-                ?: mutableMapOf()
+                ?: linkedMapOf()
         }
         set(value) {
             val list = value.map { it.value }
             settings[KEY_ACCESS_TOKENS_CACHE] =
                 json.encodeToString(ListSerializer(AccessToken.serializer()), list)
         }
+    private val memCache: LinkedHashMap<String, AccessToken> by lazy { diskCache }
 
-    private val memCache: MutableMap<String, AccessToken> by lazy { diskCache.toMutableMap() }
-
-    override fun getAccessToken(server: String): String? = memCache[server]?.token
+    override fun getAccessToken(server: String): String? =
+        lock.withLock { memCache[server]?.token }
 
-    override suspend fun saveAccessToken(server: String, token: String) {
-        memCache[server] = AccessToken(token = token, server = server)
-        diskCache = memCache
+    override fun saveAccessToken(server: String, token: String) {
```

Comment:
probably this should still be a `suspend` function. JSON encoding/decoding should probably not be done on main thread.

- [ ] Add back `suspend` for async saving of access token